app/commands/codemod/review.go

```diff
@@ -0,0 +1,365 @@
+package codemod
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"gopkg.in/yaml.v2"
+
+	"time"
+
+	"github.com/briandowns/spinner"
+	"github.com/codegangsta/cli"
+	"github.com/codelingo/lingo/app/util"
+	"github.com/codelingo/lingo/service"
+	grpcclient "github.com/codelingo/lingo/service/grpc"
+	"github.com/codelingo/platform/flow/rpc/client"
+	flow "github.com/codelingo/platform/flow/rpc/flowengine"
+
+	"context"
+
+	"github.com/juju/errors"
+)
+
+func RequestReview(ctx context.Context, req *flow.ReviewRequest) (chan *flow.Issue, chan error, error) {
+
+	// // mock request result while testing
+	// issc := make(chan *flow.Issue)
+	// errc := make(chan error)
+	// go func() {
+	// 	defer close(issc)
+	// 	defer close(errc)
+	// 	issc <- &flow.Issue{
+	// 		CtxBefore: "before",
+	// 		CtxAfter:  "after",
+	// 		LineText:  "line text",
+	// 		Comment:   "this is comment",
+	// 		Position: &flow.IssueRange{
+	// 			Start: &flow.Position{
+	// 				Filename: "main.go",
+	// 				Offset:   10,
+	// 				Line:     2,
+	// 			},
+	// 			End: &flow.Position{
+	// 				Filename: "main.go",
+	// 				Offset:   30,
+	// 				Line:     3,
+	// 			},
+	// 		},
+	// 	}
+	// }()
+	// return issc, errc, nil
+
+	conn, err := service.GrpcConnection(service.LocalClient, service.FlowServer)
+	if err != nil {
+		return nil, nil, errors.Trace(err)
+	}
+	c := client.NewFlowClient(conn)
+
+	// Create context with metadata
+	ctx, err = grpcclient.AddGcloudApiKeyToCtx(ctx)
+	if err != nil {
+		return nil, nil, errors.Trace(err)
+	}
+	ctx, err = grpcclient.AddUsernameToCtx(ctx)
+	if err != nil {
+		return nil, nil, errors.Trace(err)
+	}
+
+	issuec, errorc, err := c.Codemod(ctx, req)
+	if err != nil {
+		return nil, nil, errors.Trace(err)
+	}
+
+	return issuec, errorc, nil
+}
+
+func MakeReport(issues []*flow.Issue, format, outputFile string) (string, error) {
+	var data []byte
+	var err error
+	switch format {
+	case "json":
+		data, err = json.Marshal(issues)
+		if err != nil {
+			return "", errors.Trace(err)
+		}
+	case "json-pretty":
+		data, err = json.MarshalIndent(issues, " ", " ")
+		if err != nil {
+			return "", errors.Trace(err)
+		}
+	default:
+		return "", errors.Errorf("Unknown format %q", format)
+	}
+
+	if outputFile != "" {
+		err = ioutil.WriteFile(outputFile, data, 0775)
+		if err != nil {
+			return "", errors.Annotate(err, "Error writing issues to file")
+		}
+		return fmt.Sprintf("Done! %d issues written to %s \n", len(issues), outputFile), nil
+	}
+
+	return string(data), nil
+}
+
+// Read a .lingo file from a filepath argument
+func ReadDotLingo(ctx *cli.Context) (string, error) {
+	var dotlingo []byte
+
+	if filename := ctx.String(util.LingoFile.Long); filename != "" {
+		var err error
+		dotlingo, err = ioutil.ReadFile(filename)
+		if err != nil {
+			return "", errors.Trace(err)
+		}
+	}
+	return string(dotlingo), nil
+}
+
```

Comment:
Boolean arguments can indicate low cohesion. Consider using a separate function for each case and helper functions for repeated code. This will make each function clearer and more modular, leading to easier maintainability.